mintapi/signIn.py

```diff
@@ -38,28 +42,50 @@
 INPUT_CSS_SELECTORS_LABEL = "input_css_selectors"
 SPAN_CSS_SELECTORS_LABEL = "span_css_selectors"
 BUTTON_CSS_SELECTORS_LABEL = "button_css_selectors"
+MFA_PROMT_TEXT = "mfa_promt_text"
+PAGE_PROMT_TEXT="Text match on page that will tell us what function to use"
+PAGE_METHOD=""
 
-MFA_METHODS = [
+SIGNIN_PAGES = [
     {
-        MFA_METHOD_LABEL: MFA_VIA_SOFT_TOKEN,
-        INPUT_CSS_SELECTORS_LABEL: '#iux-mfa-soft-token-verification-code, #ius-mfa-soft-token, [data-testid="VerifySoftTokenInput"]',
-        SPAN_CSS_SELECTORS_LABEL: "",
-        BUTTON_CSS_SELECTORS_LABEL: '#ius-mfa-soft-token-submit-btn, [data-testid="VerifySoftTokenSubmitButton"]',
+        PAGE_METHOD: "password",
+        PAGE_PROMT_TEXT: "//*[text()='Enter your Intuit password']", 
+    },
+    {
+        PAGE_METHOD: "multiple",
+        PAGE_PROMT_TEXT: "//*[text()='We found your accounts']",
     },
     {
+        PAGE_METHOD: MFA_VIA_EMAIL,
+        PAGE_PROMT_TEXT: "//*[text()='Check your email']",
         MFA_METHOD_LABEL: MFA_VIA_EMAIL,
-        INPUT_CSS_SELECTORS_LABEL: "#ius-label-mfa-email-otp, #ius-mfa-email-otp-card-challenge, #ius-sublabel-mfa-email-otp",
+        INPUT_CSS_SELECTORS_LABEL: "input[pattern='[0-9]*']",
         SPAN_CSS_SELECTORS_LABEL: '[data-testid="VerifyOtpHeaderText"]',
         BUTTON_CSS_SELECTORS_LABEL: '#ius-mfa-otp-submit-btn, [data-testid="VerifyOtpSubmitButton"]',
     },
     {
+        PAGE_METHOD: MFA_VIA_SOFT_TOKEN,
+        PAGE_PROMT_TEXT: "//*[text()='Enter your verification code']",
+        MFA_METHOD_LABEL: MFA_VIA_SOFT_TOKEN,
+        INPUT_CSS_SELECTORS_LABEL: '#iux-mfa-soft-token-verification-code, #ius-mfa-soft-token, [data-testid="VerifySoftTokenInput"]',
+        SPAN_CSS_SELECTORS_LABEL: "",
+        BUTTON_CSS_SELECTORS_LABEL: '#ius-mfa-soft-token-submit-btn, [data-testid="VerifySoftTokenSubmitButton"]',
+    },
+    {
+        PAGE_METHOD: MFA_VIA_SMS,
+        PAGE_PROMT_TEXT: "//*[text()='Check your phone']",
         MFA_METHOD_LABEL: MFA_VIA_SMS,
         INPUT_CSS_SELECTORS_LABEL: "#ius-mfa-sms-otp-card-challenge, #ius-mfa-confirm-code",
         SPAN_CSS_SELECTORS_LABEL: '[data-testid="VerifyOtpHeaderText"]',
         BUTTON_CSS_SELECTORS_LABEL: '#ius-mfa-otp-submit-btn, [data-testid="VerifyOtpSubmitButton"]',
     },
+    {
+        PAGE_METHOD: "captcha",
```

Comment:
Adding support for captcha is beyond the scope of this feature branch.  This branch is focused on a better way of detecting MFAs.  While adding CAPTCHA support would be a good enhancement, it should be handled separately.

Reply:
Sounds good